src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -1109,10 +1116,67 @@ private void AddPatchNode(Patch patch, Vector2 position)
 
         patch.ApplyPatchEventVisuals(node);
 
+        var playerSpecies = patch.FindSpeciesByID(PlayerSpeciesID);
+        if (playerSpecies != null)
+        {
+            AddPlayerPopulationIndicators(patch, playerSpecies, node, position);
+        }
+
         patchNodeContainer.AddChild(node);
         nodes.Add(node.Patch, node);
     }
 
+    private void AddPlayerPopulationIndicators(Patch patch, Species playerSpecies, Control node, Vector2 position)
+    {
+        var playerPopulationIndicatorAmount = (int)Math.Ceiling(patch.GetSpeciesSimulationPopulation(playerSpecies) *
+            Constants.PLAYER_POPULATION_POPULATION_FOR_PER_INDICATOR);
+
+        var indicatorExcess = Math.Clamp(playerSpeciesPopulationIndicators.Count - playerPopulationIndicatorAmount,
+            0,
+            playerSpeciesPopulationIndicators.Count);
+
+        // Hide excess from the end of the list
+        for (int i = 0; i < indicatorExcess; ++i)
+        {
+            playerSpeciesPopulationIndicators[playerSpeciesPopulationIndicators.Count - i].Hide();
+        }
+
+        for (int i = 0; i < playerPopulationIndicatorAmount; ++i)
+        {
+            var noCached = i >= playerSpeciesPopulationIndicators.Count;
+
+            Control indicator;
+            if (noCached)
+            {
+                indicator = new TextureRect
+                {
+                    Texture = GD.Load<Texture2D>("res://assets/textures/gui/bevel/MapDotIndicator.svg"),
+                    ExpandMode = TextureRect.ExpandModeEnum.IgnoreSize,
+                    Size = new Vector2(6, 6),
+                };
+
+                populationIndicatorContainer.AddChild(indicator);
+            }
+            else
+            {
+                indicator = playerSpeciesPopulationIndicators[i];
+
+                indicator.Show();
+            }
+
+            indicator.MouseFilter = MouseFilterEnum.Ignore;
```

Comment:
This should be in the indicator creation as it doesn't seem like this value is ever updated so it wastes time to reapply the value on each loop iteration.